core/src/main/java/jenkins/BreadCrumb.java

```diff
@@ -0,0 +1,34 @@
+package jenkins;
+
+import hudson.model.ModelObject;
+import java.util.ArrayList;
+import java.util.List;
+import jenkins.model.ModelObjectWithContextMenu;
+import org.kohsuke.stapler.Ancestor;
+import org.kohsuke.stapler.Stapler;
+
+public class BreadCrumb {
+
+    public static boolean isModel(Object o) {
+        return o instanceof ModelObject;
+    }
+
+    public static boolean isModelWithContextMenu(Object o) {
+        return o instanceof ModelObjectWithContextMenu;
+    }
+
+    public List<BreadCrumbItem> generateBreadCrumbs() {
+        List<BreadCrumbItem> breadCrumbItemList = new ArrayList<>();
+        List list = Stapler.getCurrentRequest().getAncestors();
+        for( int i=list.size()-1; i>=0; i-- ) {
+            BreadCrumbItem breadCrumbItem = new BreadCrumbItem();
```

Comment:
You are not adding the item to the list?

Reply:
I just need to populate breadCrumbItem and then iterate over it in the jelly files